### PR TITLE
Modify KMS import to speed up coldstarts

### PR DESCRIPTION
**Pull request by maxday**
### What does this PR do?

- import a more specific (lighter) package
- move import logic to decrypt (this way if a lambda does not use KMS, we skip the KMS loading time as well
- unit test kms-service

### Motivation

Performance

### Testing Guidelines

Unit tests

### Types of Changes

- [ ] Bug fix
- [ ] New feature
- [ ] Breaking change
- [x] Misc (docs, refactoring, dependency upgrade, etc.)

### Check all that apply

- [x] This PR's description is comprehensive
- [ ] This PR contains breaking changes that are documented in the description
- [ ] This PR introduces new APIs or parameters that are documented and unlikely to change in the foreseeable future
- [ ] This PR impacts documentation, and it has been updated (or a ticket has been logged)
- [x] This PR's changes are covered by the automated tests
- [ ] This PR collects user input/sensitive content into Datadog
- [x] This PR passes the integration tests (ask a Datadog member to run the tests)

### CODE REVIEW

File: src/metrics/kms-service.ts [hunk 1]
```diff
@@ -1,36 +1,21 @@
...
-import { logError } from "../utils";
-
 // In order to avoid the layer adding the 40mb aws-sdk to a deployment, (which is always available
 // in the lambda environment anyway), we use require to import the sdk, and return an error if someone
 // tries to decrypt a value.
 export class KMSService {
-  private kms?: KMS;
 
-  constructor() {
+  public async decrypt(value: string): Promise<string> {
```
Comment (DarcyRaynerDD) [inline src/metrics/kms-service.ts:6]: If somebody installs the extension and uses datadog kms api key variable, decrypt will still get called even though the library doesn't end up using the key. Might be another potential optimization for clients using KMS.

File: src/metrics/kms-service.ts [hunk 2]
```diff
@@ -1,36 +1,21 @@
...
     } catch (err) {
-      logError("optional dependency aws-sdk not installed. KMS key decryption will not work", err);
-    }
-  }
-
-  public async decrypt(value: string): Promise<string> {
-    if (this.kms === undefined) {
-      // This error should only occur if the user is running this code outside of the lambda environment,
-      // (say a local development environment).
       throw Error("optional dependency aws-sdk not installed. KMS key decryption will not work");
```
Comment (DarcyRaynerDD) [inline src/metrics/kms-service.ts:18]: I think we should log original error before throwing again.

**Review by DarcyRaynerDD — APPROVED** [review]
LGTM, great find

File: src/metrics/kms-service.ts (hunk 1 above)
Comment (maxday) [reply src/metrics/kms-service.ts:7]: absolutely ! maybe in an other PR ?

File: src/metrics/kms-service.ts (hunk 2 above)
Comment (maxday) [reply src/metrics/kms-service.ts]: ✅

File: src/metrics/kms-service.ts (hunk 1 above)
Comment (DarcyRaynerDD) [reply src/metrics/kms-service.ts:8]: Sure, I don't think it has to be done right now.

**Review by DarcyRaynerDD — APPROVED** [review]
LGTM, nice tests

File: src/metrics/kms-service.ts (hunk 1 above)
Comment (maxday) [reply src/metrics/kms-service.ts:8]: OK I'll create a ticket for that

**Pull request merged**